src/index.ts

```diff
@@ -0,0 +1,52 @@
+import { bytes_to_hex, Sha256 } from "asmcrypto.js"
+import { TFileInformation, TdrFile } from "./types"
+
+export const getFileInformation: TFileInformation = async (
+  files,
+  progressFunction
+) => {
+  let processedFiles = 1
+  return await Promise.all(
+    files.map(async file => {
+      const result = {
+        checksum: await generateHash(file),
+        size: file.size,
+        lastModified: new Date(file.lastModified),
+        path: (<TdrFile>file).webkitRelativePath
+      }
+      if (progressFunction) {
+        progressFunction({ totalFiles: files.length, processedFiles })
+        processedFiles += 1
+      }
+      return result
+    })
+  )
+}
+
+const sliceToUintArray: (blob: Blob) => Promise<Uint8Array> = async blob => {
+  const fileReader = new FileReader()
+  fileReader.readAsArrayBuffer(blob)
+  return await new Promise(resolve => {
+    fileReader.onload = () => {
+      const { result } = fileReader
+      if (result instanceof ArrayBuffer) {
+        resolve(new Uint8Array(result))
+      }
+    }
+  })
+}
+export const generateHash: (file: File) => Promise<string> = async file => {
```

Comment:
At some point (probably not now) we should consider whether we should use the built-in subtle.crypto for small files. It was [quite a lot faster](https://github.com/nationalarchives/tdr-dev-documentation/blob/master/technology-considerations/frontend_checksum_options.md#results) when processing lots of small files.

Reply:
I agree, once we start testing it with lots of files, we can have a look at performance, as you say, I'll leave it for now.